src/main/java/rs/hybridit/exceptionHandler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,46 @@
+package rs.hybridit.exceptionHandler;
+
+
+
+import org.springframework.http.HttpHeaders;
+
+import java.util.Collections;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.util.WebUtils;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+	@ExceptionHandler({ IllegalStateException.class})
+
+	public final ResponseEntity<?> handleException(Exception ex, WebRequest request) {
```

Comment:
you should use IllegalStateException as exception type, not Exception.
then no need for instanceof check
fix formatting in line 19